rakelib/test.rake

```diff
@@ -1,22 +1,31 @@
+require "logstash/pluginmanager/util"
+
 namespace "test" do
-  def run_rspec(*args)
+  task "setup" do
     require "logstash/environment"
     LogStash::Environment.bundler_setup!({:without => []})
     require "rspec/core/runner"
     require "rspec"
-    RSpec::Core::Runner.run([*args])
   end
 
-  task "core" do
-    run_rspec(Rake::FileList["spec/**/*_spec.rb"])
+  desc "run core specs"
+  task "core" => ["setup"] do
+    RSpec::Core::Runner.run([Rake::FileList["spec/**/*_spec.rb"]])
```

Comment:
@colinsurprenant please reapply the same exit code enforcement done in https://github.com/elastic/logstash/pull/2920 otherwise a failed "rake test" task will have an exit code of 0:

```
joaoduarte@Joaos-MBP-2 ~/projects/logstash (git)-[pr/2928] % bundle exec rake test:core
Using Accessor#strict_set for specs
Run options: exclude {:redis=>true, :socket=>true, :performance=>true, :couchdb=>true, :elasticsearch=>true, :elasticsearch_secure=>true, :broken=>true, :export_cypher=>true, :integration=>true, :windows=>true}
............................................................................................................................................................................................................................................F.........................................

Failures:

  1) conditionals in filter simple "{"foo":"bar"}" when processed
     Failure/Error: Unable to find matching line from backtrace
       
       expected: "awesom"
            got: "awesome"
       
       (compared using ==)
     # ./spec/core/conditionals_spec.rb:66:in `(root)'
     # ./rakelib/test.rake:13:in `(root)'

Finished in 3.61 seconds
278 examples, 1 failure

Failed examples:

rspec ./vendor/bundle/jruby/1.9/gems/logstash-devutils-0.0.10-java/lib/logstash/devutils/rspec/logstash_helpers.rb:51 # conditionals in filter simple "{"foo":"bar"}" when processed

Randomized with seed 18354

joaoduarte@Joaos-MBP-2 ~/projects/logstash (git)-[pr/2928] % echo $?
0
```